### PR TITLE
Train for 1.9.1

### PR DESCRIPTION
**Pull request by amitaekbote**
## High Level Description

#1531 
#1536 
#1529 
#1538 

## Checklist for all PR's

  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [ ] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [ ] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]

### CODE REVIEW

**Comment by amitaekbote** [comment]
Depends on #1536 getting two reviews, optimizing testing time

File: packages/dcos-metrics/buildinfo.json [hunk 1]
```diff
@@ -3,8 +3,8 @@
   "single_source" : {
     "kind": "git",
     "git": "https://github.com/dcos/dcos-metrics.git",
-    "ref": "b68c5e128120d0bf96a13aa11000999170a33ed6",
-    "ref_origin": "1.1.0"
+    "ref": "62702c3396deb5b47f4492a5a39d9dfb6f95ab0b",
+    "ref_origin": "master"
```
Comment (adam-mesos) [inline packages/dcos-metrics/buildinfo.json]: @collinvandyck @klueska Are we sure we want to use `master` of `dcos-metrics` in 1.9.1? Have there been other significant changes between these two SHAs? I see 64 commits, and they can't all be the backport for DCOS_OSS-925
https://github.com/dcos/dcos-metrics/compare/b68c5e128120d0bf96a13aa11000999170a33ed6...62702c3396deb5b47f4492a5a39d9dfb6f95ab0b
https://github.com/dcos/dcos/pull/1527 is targeted at master/1.10, so we'll probably need a different PR for 1.9.1 that actually shows the component changelog in the PR description.

File: packages/spartan/buildinfo.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
     "spartan": {
       "kind": "git",
       "git": "https://github.com/dcos/spartan.git",
-      "ref": "0b5243b8032bf475fb2c62ec4f92142d701f35ad",
+      "ref": "f7b2311c71fa833c182a13781af00a387632fb1b",
```
Comment (adam-mesos) [inline packages/spartan/buildinfo.json]: https://github.com/dcos/spartan/compare/0b5243b8032bf475fb2c62ec4f92142d701f35ad...f7b2311c71fa833c182a13781af00a387632fb1b looks quite specific to the single ip-blacklist/forward-zone feature described in https://github.com/dcos/dcos/pull/1536 which is properly targeted to 1.9. TL;DR: The SHA-bump looks fine.

However, I'm not happy using the `master` ref of components in a patch release. I'd prefer that we branch/version spartan, perhaps with a branch name (e.g. `1.9.1`) to match the DC/OS release it's going out with. Or at least a `1.9.x` vs. `1.10.x` so we can backport fixes to different branches for patch releases. Maybe we don't have a need to yet, if spartan development is pretty quiet, but the day will come. @nlsun @drewkerrigan

File: packages/dcos-metrics/buildinfo.json (hunk 1 above)
Comment (collinvandyck) [inline packages/dcos-metrics/buildinfo.json]: @adam-mesos I created a `1.9.1` tag on dcos-metrics which only has the changes for `DCOS_OSS-925`.   The ref is `a01acda5647c3108cff62851bc694d26d1a503d5`.  I'm not sure if we need to backport any other fixes in dcos-metrics:master into 1.9.1. -- I'll leave that up to @klueska to decide.

**Comment by amitaekbote** [comment]
Good catch @adam-mesos, I bumped it up and it should not include all the changes for metrics

**Comment by orsenthil** [comment]
The PR template needs to be updated. Please update the PR template before merging this in.

**Comment by adam-mesos** [comment]
Happy with the new metrics bump, but we still need a downstream train and some investigation into the test failures.

File: packages/spartan/buildinfo.json (hunk 2 above)
Comment (nlsun) [inline packages/spartan/buildinfo.json]: Ok, will branch.

Comment (nlsun) [inline packages/spartan/buildinfo.json]: @adam-mesos changed the ref from "master" to "dcos/1.9.1"

Comment (adam-mesos) [inline packages/spartan/buildinfo.json]: @amitaekbote Can you update this train with the new ref from #1536 ?
Either remove the old commits and re-add the new ones, or just change the ref to `dcos/1.9.1` if that's all that changed.

**Comment by darkonie** [comment]
@adam-mesos looks like the tests failed due to flakiness and not actual change, rerunning now...

**Comment by nlsun** [comment]
currently changing the way that spartan was branched

**Comment by nlsun** [comment]
ok, done changing spartan, train should be updated with new ref and commit

File: packages/spartan/buildinfo.json (hunk 2 above)
Comment (adam-mesos) [inline packages/spartan/buildinfo.json]: @nlsun Please confirm that ref should be https://github.com/dcos/spartan/commit/e94d5661cd271387379fd41001b80a31a1b4eab8 from branch https://github.com/dcos/spartan/commits/dcos/1.9
I diffed the 1.9 branch against master against `nlsun/backport-blacklist`, and they've all got the same bits, even if different SHAs.

**Pull request merged**